Description: Upstream changes introduced in version 3.0.0~rc1-1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 php-imagick (3.0.0~rc1-1) unstable; urgency=low
 .
   * New upstream release (Closes: #524863, #570649, #573139)
   * New maintainer (Closes: #530621)
   * Change Build-Depends from libmagick9-dev to libmagickwand-dev
     (Thanks Ivan Borzenkov)
   * debian/rules:
     + Rewrite php5 oriented only, work in progress to adapt multi php version
     + Remove tarball mode rules
     + Fix generated path of debian/post{inst,rm} in clean rule
     + Delete "--with-regex" and "--disable-rpath" options (enabled by default)
       in configure call (unrecognized options)
     + Delete lintian override (no-shlibs-control-file)
     + Touch examples/ and CREDITS in upstream tarball
       (lintian package-contains-ancient-file rule)
     + Delete unnecessary patch rule
     + phpize5 --clean is now called in clean rule
     + Set CFLAGS to -Wl,--as-needed, to not link against some libX11
   * Bump to Standards-Version 3.8.4 (move section from web to php)
   * Bump to compatibility 7 in debian/{control,compat}
   * debian/phpX-imagick.post{inst,rm}:
     + add "set -e" (instead of "#!/bin/sh -e")
     + comment loading on remove (Closes: #498150)
     (Thanks Ivan Borzenkov)
   * debian/imagick.ini: change comment style from "#" to ";"
     (Closes: #563433, #566970, thanks Thijs Kinkhorst)
   * debian/{changelog,copyright}: suppress unnecessary spaces
   * Switch to dpkg-source 3.0 (quilt) format
   * debian/watch: mangle version according Debian Policy 5.6.12
   * debian/ltmain.patch: Patch to reorder linker argument after phpize
     (From BTS#347650)
 .
 The person named in the Author field signed this changelog entry.
Author: Guillaume Delacour <gui@iroqwa.org>
Bug-Debian: http://bugs.debian.org/498150
Bug-Debian: http://bugs.debian.org/524863
Bug-Debian: http://bugs.debian.org/530621
Bug-Debian: http://bugs.debian.org/563433
Bug-Debian: http://bugs.debian.org/566970
Bug-Debian: http://bugs.debian.org/570649
Bug-Debian: http://bugs.debian.org/573139

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- /dev/null
+++ php-imagick-3.0.0~rc1/z
@@ -0,0 +1,164 @@
+diff -Nur ../../new/php-imagick-3.0.0~rc1/debian/changelog debian/changelog
+--- ../../new/php-imagick-3.0.0~rc1/debian/changelog	2010-05-31 19:57:50.000000000 +0200
++++ debian/changelog	2010-06-01 22:45:33.000000000 +0200
+@@ -15,6 +15,7 @@
+       (lintian package-contains-ancient-file rule)
+     + Delete unnecessary patch rule
+     + phpize5 --clean is now called in clean rule
++    + Set CFLAGS to -Wl,--as-needed, to not link against some libX11
+   * Bump to Standards-Version 3.8.4 (move section from web to php)
+   * Bump to compatibility 7 in debian/{control,compat}
+   * debian/phpX-imagick.post{inst,rm}:
+@@ -26,6 +27,8 @@
+   * debian/{changelog,copyright}: suppress unnecessary spaces
+   * Switch to dpkg-source 3.0 (quilt) format
+   * debian/watch: mangle version according Debian Policy 5.6.12
++  * debian/ltmain.patch: Patch to reorder linker argument after phpize
++    (From BTS#347650)
+ 
+  -- Guillaume Delacour <gui@iroqwa.org>  Tue, 09 Mar 2010 17:58:50 +0000
+ 
+diff -Nur ../../new/php-imagick-3.0.0~rc1/debian/ltmain.patch debian/ltmain.patch
+--- ../../new/php-imagick-3.0.0~rc1/debian/ltmain.patch	1970-01-01 01:00:00.000000000 +0100
++++ debian/ltmain.patch	2010-06-01 21:24:23.000000000 +0200
+@@ -0,0 +1,30 @@
++--- ltmain.sh.old	2010-06-01 19:21:42.000000000 +0000
+++++ ltmain.sh	2010-06-01 19:23:14.000000000 +0000
++@@ -4716,6 +4716,11 @@
++ 	arg=$func_stripname_result
++ 	;;
++ 
+++      -Wl,--as-needed)
+++	deplibs="$deplibs $arg"
+++	continue
+++	;;
+++
++       -Wl,*)
++ 	func_stripname '-Wl,' '' "$arg"
++ 	args=$func_stripname_result
++@@ -5070,6 +5075,15 @@
++ 	lib=
++ 	found=no
++ 	case $deplib in
+++	-Wl,--as-needed)
+++	  if test "$linkmode,$pass" = "prog,link"; then
+++	    compile_deplibs="$deplib $compile_deplibs"
+++	    finalize_deplibs="$deplib $finalize_deplibs"
+++	  else
+++	    deplibs="$deplib $deplibs"
+++	  fi
+++	  continue
+++	  ;;
++ 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
++ 	  if test "$linkmode,$pass" = "prog,link"; then
++ 	    compile_deplibs="$deplib $compile_deplibs"
+diff -Nur ../../new/php-imagick-3.0.0~rc1/debian/patches/debian-changes-3.0.0~rc1-1 debian/patches/debian-changes-3.0.0~rc1-1
+--- ../../new/php-imagick-3.0.0~rc1/debian/patches/debian-changes-3.0.0~rc1-1	2010-06-01 01:43:48.000000000 +0200
++++ debian/patches/debian-changes-3.0.0~rc1-1	1970-01-01 01:00:00.000000000 +0100
+@@ -1,68 +0,0 @@
+-Description: Upstream changes introduced in version 3.0.0~rc1-1
+- This patch has been created by dpkg-source during the package build.
+- Here's the last changelog entry, hopefully it gives details on why
+- those changes were made:
+- .
+- php-imagick (3.0.0~rc1-1) unstable; urgency=low
+- .
+-   * New upstream release (Closes: #524863, #570649, #573139)
+-   * New maintainer (Closes: #530621)
+-   * Change Build-Depends from libmagick9-dev to libmagickwand-dev
+-     (Thanks Ivan Borzenkov)
+-   * debian/rules:
+-     + Rewrite php5 oriented only, work in progress to adapt multi php version
+-     + Remove tarball mode rules
+-     + Fix generated path of debian/post{inst,rm} in clean rule
+-     + Delete "--with-regex" and "--disable-rpath" options (enabled by default)
+-       in configure call (unrecognized options)
+-     + Delete lintian override (no-shlibs-control-file)
+-     + Touch examples/ and CREDITS in upstream tarball
+-       (lintian package-contains-ancient-file rule)
+-     + Delete unnecessary patch rule
+-     + phpize5 --clean is now called in clean rule
+-   * Bump to Standards-Version 3.8.4 (move section from web to php)
+-   * Bump to compatibility 7 in debian/{control,compat}
+-   * debian/phpX-imagick.post{inst,rm}:
+-     + add "set -e" (instead of "#!/bin/sh -e")
+-     + comment loading on remove (Closes: #498150)
+-     (Thanks Ivan Borzenkov)
+-   * debian/imagick.ini: change comment style from "#" to ";"
+-     (Closes: #563433, #566970, thanks Thijs Kinkhorst)
+-   * debian/{changelog,copyright}: suppress unnecessary spaces
+-   * Switch to dpkg-source 3.0 (quilt) format
+-   * debian/watch: mangle version according Debian Policy 5.6.12
+- .
+- The person named in the Author field signed this changelog entry.
+-Author: Guillaume Delacour <gui@iroqwa.org>
+-Bug-Debian: http://bugs.debian.org/498150
+-Bug-Debian: http://bugs.debian.org/524863
+-Bug-Debian: http://bugs.debian.org/530621
+-Bug-Debian: http://bugs.debian.org/563433
+-Bug-Debian: http://bugs.debian.org/566970
+-Bug-Debian: http://bugs.debian.org/570649
+-Bug-Debian: http://bugs.debian.org/573139
+-
+----
+-The information above should follow the Patch Tagging Guidelines, please
+-checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
+-are templates for supplementary fields that you might want to add:
+-
+-Origin: <vendor|upstream|other>, <url of original patch>
+-Bug: <url in upstream bugtracker>
+-Bug-Debian: http://bugs.debian.org/<bugnumber>
+-Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
+-Forwarded: <no|not-needed|url proving that it has been forwarded>
+-Reviewed-By: <name and email of someone who approved the patch>
+-Last-Update: <YYYY-MM-DD>
+-
+---- php-imagick-3.0.0~rc1.orig/imagick-3.0.0RC1/config.m4
+-+++ php-imagick-3.0.0~rc1/imagick-3.0.0RC1/config.m4
+-@@ -84,7 +84,7 @@ if test $PHP_IMAGICK != "no"; then
+- 
+-   IMAGICK_LIBS=`$WAND_BINARY --libs`
+-   IMAGICK_INCS=`$WAND_BINARY --cflags`
+--      
+-+
+-   PHP_EVAL_LIBLINE($IMAGICK_LIBS, IMAGICK_SHARED_LIBADD)
+-   PHP_EVAL_INCLINE($IMAGICK_INCS)
+- 
+diff -Nur ../../new/php-imagick-3.0.0~rc1/debian/patches/series debian/patches/series
+--- ../../new/php-imagick-3.0.0~rc1/debian/patches/series	2010-06-01 00:58:07.000000000 +0200
++++ debian/patches/series	1970-01-01 01:00:00.000000000 +0100
+@@ -1 +0,0 @@
+-debian-changes-3.0.0~rc1-1
+diff -Nur ../../new/php-imagick-3.0.0~rc1/debian/rules debian/rules
+--- ../../new/php-imagick-3.0.0~rc1/debian/rules	2010-06-01 01:43:38.000000000 +0200
++++ debian/rules	2010-06-01 22:45:50.000000000 +0200
+@@ -6,11 +6,12 @@
+ 
+ PHP_EX5=$(shell /usr/bin/php-config5 --extension-dir)
+ 
+-LDFLAGS := -Wl,--as-needed $(LDFLAGS)
+-
+ DEB_INSTALL_EXAMPLES_php5-imagick = $(DEB_SRCDIR)/examples/*
+ DEB_INSTALL_DOCS_php5-imagick = $(DEB_SRCDIR)/CREDITS
+ 
++# don't link against unnecessary libraries such as libX11
++LDFLAGS := -Wl,--as-needed $(LDFLAGS)
++
+ #
+ # local hacks
+ #
+@@ -21,8 +22,13 @@
+ 	cd $(DEB_SRCDIR) && phpize5 --clean
+ 
+ configure_for_php5::
++	# Hack here because:
++	# - we don't want the module linked against some libx* (unused symbols)
++	# We want to link with just needed libraries
++	# - "-Wl,--as-needed" is ignored if not the first argument of the linker
++	# We have to patch the linker call script ltmain.sh
+ 	cd $(DEB_SRCDIR) && phpize5 && \
+-	    patch -p1 </tmp/libgnome-2.30.0/debian/patches/00_as-needed.patch && \
++	    patch -p1 < ../debian/ltmain.patch ltmain.sh && \
+ 	    ./configure --with-imagick --with-php-config=/usr/bin/php-config5 \
+ 	    --disable-static --with-imagick=shared,/usr
+ 	sed -e 's/phpX/php5/g' < debian/phpX-imagick.postinst > debian/php5-imagick.postinst
